packages/repository-json-schema/src/build-schema.ts

```diff
@@ -436,7 +436,19 @@ export function modelToJsonSchema<T extends object>(
 
     const propSchema = getJsonSchema(referenceType, options);
 
-    includeReferencedSchema(referenceType.name, propSchema);
+    if (typeof result.properties[p] !== 'boolean') {
+      const prop = result.properties[p] as JSONSchema;
+      const propTitle = propSchema.title ?? referenceType.name;
+      const targetRef = {$ref: `#/definitions/${propTitle}`};
```

Comment:
I've observed that sometime people put values in title that don't align with a type name:

```
 async create(
    @requestBody({
      content: {
        'application/json': {
          schema: getModelSchemaRef(NewUserRequest, {
            title: 'NewUser',
            exclude: ['id'],
          }),
        },
      },
    })
```
so could something errorneous happen here using `propTitle`?

Reply:
That's fine as long as the title is unique. Please note we add `Partial` suffix to schemas built with `{partial: true}`.